contracts/strategies/harvest/IdleHarvestStrategy.sol

```diff
@@ -0,0 +1,240 @@
+
+    /// @notice underlying ERC20 token contract
+    IERC20Detailed public underlyingToken;
+
+    /// @notice address of the reward pool
+    address public rewardPool;
+
+    /// @notice amount last indexed for calculating APR
+    uint256 public lastIndexAmount;
+
+    /// @notice time when last deposit/redeem was made, used for calculating the APR
+    uint256 public lastIndexedTime;
+
+    /// @notice address of the IdleCDO
+    address public idleCDO;
+
+    /// @notice one year, used to calculate the APR
+    uint256 public constant YEAR = 365 days;
+
+    /// @notice address of the governance token. (Here FARM)
+    address public govToken;
+
+    /// @notice uniswap router path that should be used to swap the tokens
+    address[] public uniswapRouterPath;
+
+    /// @notice interface derived from uniswap router
+    IUniswapV2Router02 public uniswapV2Router02;
+
+    /// @notice latest saved apr
+    uint256 public lastApr;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        token = address(1);
+    }
+
+    /// @notice can be only called once
+    /// @param _strategyToken address of the strategy token
+    /// @param _underlyingToken address of the underlying token
+    /// @param _rewardPool address of the reward pool
+    /// @param _uniswapV2Router02 address of the uniswap router
+    /// @param _routerPath path to swap governance tokens
+    function initialize(
+        address _strategyToken,
+        address _underlyingToken,
+        address _rewardPool,
+        address _uniswapV2Router02,
+        address[] calldata _routerPath,
+        address _owner
+    ) public initializer {
+        OwnableUpgradeable.__Ownable_init();
+        ReentrancyGuardUpgradeable.__ReentrancyGuard_init();
+        require(token == address(0), "Token is already initialized");
+
+        //----- // -------//
+        strategyToken = _strategyToken;
+        rewardPool = _rewardPool;
+        token = _underlyingToken;
+        underlyingToken = IERC20Detailed(token);
+        tokenDecimals = underlyingToken.decimals();
+        oneToken = 10**(tokenDecimals);
+
+        govToken = IRewardPool(rewardPool).rewardToken();
+
+        ERC20Upgradeable.__ERC20_init("Idle Harvest Strategy Token", string(abi.encodePacked("idleHS", underlyingToken.symbol())));
+        //------//-------//
+
+        uniswapV2Router02 = IUniswapV2Router02(_uniswapV2Router02);
+        uniswapRouterPath = _routerPath;
+
+        transferOwnership(_owner);
+        lastIndexedTime = block.timestamp;
+    }
+
+    /// @notice redeem the rewards. Claims reward as per the _extraData
+    /// @return rewards amount of reward that is deposited to vault
+    function redeemRewards(bytes calldata) external override onlyIdleCDO returns (uint256[] memory rewards) {
+        rewards = _redeemRewards();
+    }
+
+    /// @notice internal function to claim the rewards
+    function _redeemRewards() internal returns (uint256[] memory rewards) {
```

Comment:
I would merge the `redeemRewards` and `_redeemRewards` into a single method given that it's used only in the external `redeemRewards` now